lib/house.rb

```diff
@@ -0,0 +1,52 @@
+class House
+attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    rooms.push(room)
+  end
+
+  def price_to_i
+    price.delete! "$"
+    price.to_i
+  end
+
+  def above_market_average?
+    if price_to_i >= 500000
+      true
+    else
+      false
+    end
+  end
+
+  def rooms_from_category(category)
+    category_rooms = []
+    rooms.find_all do |room|
+      if room.category == category
+        category_rooms << room
+      end
+    end
+    category_rooms
+  end
```

Comment:
This works, but you are using the `each` pattern with `find_all`.  either replace your find_all with `each` or:

```ruby
  def rooms_from_category(category)
    @rooms.find_all do |room|
      room.category == category
    end
  end
```